### ISSUE TITLE
Importing the package without a bundler is broken

### ISSUE THREAD

**Issue by andreashuetter** [alonrbar/easy-template-x#120]
Is the latest version 4.1.0 somehow broken for ES6 support, and does perhaps the demo still use an older version?!
I downloaded the original docx template file from the demo.

With this code:
```
import { TemplateHandler } from 'easy-template-x';
import * as fs from 'fs';
const templateFile = fs.readFileSync('./template.docx');
const data = {
  "Beers": [
    { "Brand": "Carlsberg", "Price": 1 },
    { "Brand": "Leaf Blonde", "Price": 2 },
    { "Brand": "Weihenstephan", "Price": 1.5 }
  ]
};
const handler = new TemplateHandler();
const doc = await handler.process(templateFile, data);
fs.writeFileSync('./output.docx', doc);
```
I get this error:
```
import { TemplateHandler } from 'easy-template-x';
         ^^^^^^^^^^^^^^^
SyntaxError: The requested module 'easy-template-x' does not provide an export named 'TemplateHandler'
    at ModuleJob._instantiate (node:internal/modules/esm/module_job:132:21)
    at async ModuleJob.run (node:internal/modules/esm/module_job:214:5)
    at async ModuleLoader.import (node:internal/modules/esm/loader:329:24)
    at async loadESM (node:internal/process/esm_loader:28:7)
    at async handleMainPromise (node:internal/modules/run_main:113:12)
```

So i changed the import statement in the first line to this instead:
`import { TemplateHandler } from './node_modules/easy-template-x/dist/es/easy-template-x.js';`

Now I'm getting this:
```
MalformedFileError: Malformed file detected. Make sure the file is a valid docx file.
    at TemplateHandler.loadDocx (file:///home/andreas/nodejs/app/printing_template/node_modules/easy-template-x/dist/es/easy-template-x.js:3226:13)
    at async TemplateHandler.process (file:///home/andreas/nodejs/app/printing_template/node_modules/easy-template-x/dist/es/easy-template-x.js:3136:18)
    at async file:///home/andreas/nodejs/app/printing_template/printing_template.js:32:13 {
  expectedFileType: 'docx'
}
```

But when I install the older version 3.0.0 of easy-template-x then everything works without any problems (without changing the import statement, and with the same docx template file).

----

**_Edit: The files are in fact not the same!_**
I got to run it by downloading the sandbox from https://codesandbox.io/p/sandbox/easy-template-x-demo-x4ppu and modifying my code to this:
```
const { TemplateHandler } = require('./sandbox/dist/cjs/easy-template-x.js');
const fs = require('fs');
const templateFile = fs.readFileSync('./template.docx');
const data = {
  "Beers": [
    { "Brand": "Carlsberg", "Price": 1 },
    { "Brand": "Leaf Blonde", "Price": 2 },
    { "Brand": "Weihenstephan", "Price": 1.5 }
  ]
};
(async() => {
  const handler = new TemplateHandler();
  const doc = await handler.process(templateFile, data);
  fs.writeFileSync('./output.docx', doc);
})();
```

But i cannot get it to run with the latest files from Github (neither the cjs nor the es version).
It seems to me that the Github version that I get through npm is erroneous.

Performing a diff between the dist/cjs from Github and the dist/cjs from the sandbox showed a lot of differences.

**Comment by alonrbar** [comment]
Hi, thank you for reporting this. I'll take a look.

**Comment by alonrbar** [comment]
So, from what I understand the issue was a package resolution issue. I think it was related to the yarn configurations. 
Anyway, I've updated the demo and also added a README.md file to it with a step by step instructions for running it locally. Please try to download the updated demo to a new folder and let me know how it goes now.

**Comment by alonrbar** [comment]
The example uses an es module syntax, so you either need to specify ["type": "module"](https://nodejs.org/api/esm.html#introduction) in your package.json file or use a bundler (the demo uses Typescript with Parcel, on my personal projects I usually use Typescript with Webpack). Alternatively, you can change the import statement to a require statement. This will look something like this:
```js
const easy = require("easy-template-x");

const handler = new easy.TemplateHandler();
```

**Comment by andreashuetter** [comment]
> The example uses an es module syntax, so you either need to specify ["type": "module"](https://nodejs.org/api/esm.html#introduction) in your package.json file or use a bundler (the demo uses Typescript with Parcel, on my personal projects I usually use Typescript with Webpack). Alternatively, you can change the import statement to a require statement. This will look something like this:
> 
> ```js
> const easy = require("easy-template-x");
> 
> const handler = new easy.TemplateHandler();
> ```

Yes I know. But the code from the Github Repo does not work, neither as CommonJS (using "require") nor as ES Module (specifying "type": "module" in package.json and then using import).

First, I do a fresh npm install --save easy-template-x to get the latest version (4.1.0) from the github repo.

Then I create to different projects for testing, one as CommonJS (without "type": "module" in package.json) and as ESM with "type": "module" in package.json):

**1. Common:**
```
const TemplateHandler = require('easy-template-x');
const fs = require('fs');
const templateFile = fs.readFileSync('./template.docx');
const data = {
  "Beers": [
    { "Brand": "Carlsberg", "Price": 1 },
    { "Brand": "Leaf Blonde", "Price": 2 },
    { "Brand": "Weihenstephan", "Price": 1.5 }
  ]
};
(async() => {
        const handler = new TemplateHandler();
        const doc = await handler.process(templateFile, data);

        fs.writeFileSync('./output.docx', doc);
})();
```
```
node test_cjs.js
/opt/app_sam_dev/home/ha016863/nodejs/app/test_cjs/test_cjs.js:1
const TemplateHandler = require('easy-template-x');
                        ^

Error [ERR_REQUIRE_ESM]: require() of ES Module /home/andreas/nodejs/app/test_cjs/node_modules/easy-template-x/dist/cjs/easy-template-x.js from /opt/app_sam_dev/home/ha016863/nodejs/app/test_cjs/test_cjs.js not supported.
easy-template-x.js is treated as an ES module file as it is a .js file whose nearest parent package.json contains "type": "module" which declares all .js files in that package scope as ES modules.
Instead either rename easy-template-x.js to end in .cjs, change the requiring code to use dynamic import() which is available in all CommonJS modules, or change "type": "module" to "type": "commonjs" in /home/andreas/nodejs/app/test_cjs/node_modules/easy-template-x/package.json to treat all .js files as CommonJS (using .mjs for all ES modules instead).

    at Object.<anonymous> (/opt/app_sam_dev/home/ha016863/nodejs/app/test_cjs/test_cjs.js:1:25) {
  code: 'ERR_REQUIRE_ESM'
}
```


**2. ESM:**
adding the line

"type": "module" to package.json and then running this code:

```
import { TemplateHandler } from 'easy-template-x';
import fs from 'fs';
const templateFile = fs.readFileSync('./template.docx');
const data = {
  "Beers": [
    { "Brand": "Carlsberg", "Price": 1 },
    { "Brand": "Leaf Blonde", "Price": 2 },
    { "Brand": "Weihenstephan", "Price": 1.5 }
  ]
};
(async() => {
        const handler = new TemplateHandler();
        const doc = await handler.process(templateFile, data);

        fs.writeFileSync('./output.docx', doc);
})();
```
 node test_esm.js
file:///opt/app_sam_dev/home/ha016863/nodejs/app/test_esm/test_esm.js:1
import { TemplateHandler } from 'easy-template-x';
         ^^^^^^^^^^^^^^^
SyntaxError: The requested module 'easy-template-x' does not provide an export named 'TemplateHandler'
    at ModuleJob._instantiate (node:internal/modules/esm/module_job:132:21)
    at async ModuleJob.run (node:internal/modules/esm/module_job:214:5)
    at async ModuleLoader.import (node:internal/modules/esm/loader:329:24)
    at async loadESM (node:internal/process/esm_loader:28:7)
    at async handleMainPromise (node:internal/modules/run_main:113:12)
```
Because this isn't working, I tried the following: Changing the import statement in the first line, so we have this:
```
import { TemplateHandler } from './node_modules/easy-template-x/dist/es/easy-template-x.js';
import fs from 'fs';
const templateFile = fs.readFileSync('./template.docx');
const data = {
  "Beers": [
    { "Brand": "Carlsberg", "Price": 1 },
    { "Brand": "Leaf Blonde", "Price": 2 },
    { "Brand": "Weihenstephan", "Price": 1.5 }
  ]
};
(async() => {
        const handler = new TemplateHandler();
        const doc = await handler.process(templateFile, data);

        fs.writeFileSync('./output.docx', doc);
})();
```
But thus I am getting this error, even when using the same template.docx file that works in the demo:
```
 node test_esm.js
file:///opt/app_sam_dev/home/ha016863/nodejs/app/test_esm/node_modules/easy-template-x/dist/es/easy-template-x.js:3226
      throw new MalformedFileError('docx');
            ^

MalformedFileError: Malformed file detected. Make sure the file is a valid docx file.
    at TemplateHandler.loadDocx (file:///opt/app_sam_dev/home/ha016863/nodejs/app/test_esm/node_modules/easy-template-x/dist/es/easy-template-x.js:3226:13)
    at async TemplateHandler.process (file:///opt/app_sam_dev/home/ha016863/nodejs/app/test_esm/node_modules/easy-template-x/dist/es/easy-template-x.js:3136:18)
    at async file:///opt/app_sam_dev/home/ha016863/nodejs/app/test_esm/test_esm.js:13:21 {
  expectedFileType: 'docx'
}
```

**Comment by alonrbar** [comment]
OK, so using the package _without a bundler_ is indeed broken. I'll issue a fix soon. Thank you for raising this.

In case you are interested about the details:

* The cjs version is broken since the `easy-template-x` package itself is now using `"type": "module"`. I need to make the files in the `dist/cjs` dir have the cjs extension. 

* The esm version is broken because typescript and esm has slightly different import semantics and loading of the JSZip dependency gets broken.

**Comment by alonrbar** [comment]
Published v4.1.1 with the fix. Thanks again :)

**Comment by alonrbar** [comment]
Maybe it has something to do with the Node or Typescript versions used 🤔 Can you write down your versions?

**Comment by tadmi** [comment]
Start from 4.1.1 have a problem with types (on 4.1.0 OK). Node 23, Typescript 5.6.3.
![image](https://github.com/user-attachments/assets/08c4c134-1fdd-4098-855f-55226090395d)

**Comment by tadmi** [comment]
Editing the package.json returns types for ESM

![image](https://github.com/user-attachments/assets/389b9aa4-7307-45d9-b00b-b47ce6ca075d)

**Comment by alonrbar** [comment]
@tadmi & @GautierT 
Published 4.1.2 which should resolve the issue. Thank you for your help!
